common/src/main/java/io/netty/util/ConstantPool.java

```diff
@@ -68,7 +68,7 @@ public T valueOf(String name) {
     private T getOrCreate(String name) {
         T constant = constants.get(name);
         if (constant == null) {
-            final T tempConstant = newConstant(nextId(), name);
+            final T tempConstant = newConstant(0, name);
```

Comment:
@Scottmitch new method will brake existing inheritance as every child need to override it. Not sure why you highlighted exactly this line as this line guaranteed unique id for constant so exactly in this case semantics not chnaged.